### PR TITLE
#56: completion bash only

### PR DESCRIPTION
**Pull request by hohwille**
Replacement for PR #178
Without jline and completion shell (that will come later in a separate PR).

### CODE REVIEW

File: cli/src/test/java/com/devonfw/tools/ide/completion/CompleteTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  /** Test of {@link AbstractIdeContext#complete(CliArguments, boolean) auto-completion} for input "help", "". */
+  @Test
+  public void testCompleteHelpEmptyArgs() {
+
+    // arrange
+    AbstractIdeContext context = IdeTestContextMock.get();
+    CliArguments args = CliArguments.ofCompletion("help", "");
+    List<String> expectedCandidates = getExpectedCandidates(context, true, false);
+    expectedCandidates.remove("-v"); // hackish solution, improve me please
```
Comment (jan-vcapgemini) [inline cli/src/test/java/com/devonfw/tools/ide/completion/CompleteTest.java:102]: Maybe add a TODO for this.

**Review by jan-vcapgemini — CHANGES_REQUESTED** [review]
Thanks for this isolated autocompletion PR. I really like the new Completion tests :)
I've added some small CR's.

**Review by jan-vcapgemini — APPROVED** [review]
Looks fine to me now.

**Review by jan-vcapgemini — APPROVED** [review]
Ready for merge.

**Comment by hohwille** [comment]
> Ready for merge.

Thanks @jan-vcapgemini for the review and the pro-active improvements. 👍 
I just merged now.

**Pull request merged**